lib/membrane/core/parent/child_life_controller/link_handler.ex

```diff
@@ -195,4 +189,14 @@ defmodule Membrane.Core.Parent.ChildLifeController.LinkHandler do
     ])
     ~> {&1, state}
   end
+
+  defp send_linking_finished(links) do
+    endpoints =
+      links
+      |> Enum.flat_map(&[&1.from, &1.to])
+      |> Enum.uniq()
+      |> Enum.reject(&(&1.child == {Membrane.Bin, :itself}))
```

Comment:
Probably negligible performance impact, but `uniq` is probably `O(n*log(n))`:
```suggestion
      links
      |> Enum.flat_map(&[&1.from, &1.to])
      |> Enum.reject(&(&1.child == {Membrane.Bin, :itself}))
      |> Enum.uniq()
```